### PR TITLE
Feature/15 admin dashboard

### PR DESCRIPTION
**Pull request by timohuber**
- Add location statistics
- Show lists on incomplete and upcoming sessions on admin dashboard
- Include experiment in session type
- Add total emails sent to pool statistics

### CODE REVIEW

File: pool/app/pool_location/pool_location.ml [hunk 1]
```diff
@@ -11,3 +11,16 @@ let find_location_file = Repo_file_mapping.find
...
+
+module Statistics = struct
+  include Entity_statistics
+
+  let current_year () = Ptime_clock.now () |> Ptime.to_year
+  let create ?(year = current_year ()) = Repo_statistics.statistics year
+
+  let year_select () =
+    let num_years = 10 in
+    let current_year = current_year () in
```
Comment (mabiede) [inline pool/app/pool_location/pool_location.ml:23]: As discussed: `SELECT COALESCE(created_at, NOW()) FROM pool_assignments ORDER BY created_at DESC LIMIT 1`

File: pool/app/session/repo/repo.ml [hunk 2]
```diff
@@ -844,12 +850,49 @@ module Sql = struct
...
+      AND pool_sessions.canceled_at IS NULL
+      AND (pool_sessions.start + INTERVAL duration SECOND) < NOW()
+    |sql}
+    |> query_by_admin
+  ;;
+
+  let find_upcoming_by_admin =
+    {sql|
+      (pool_sessions.start + INTERVAL duration SECOND) > NOW()
+      AND pool_sessions.closed_at IS NULL
```
Comment (mabiede) [inline pool/app/session/repo/repo.ml:882]: guess, `canceled_at IS NULL` should be in here as well

File: pool/web/handler/admin_dashboard.ml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  module Statistics : sig
+    val read : Rock.Middleware.t
+  end
+end = struct
+  module Guardian = Middleware.Guardian
+
+  module Statistics = struct
+    let read = Statistics.Guard.Access.read |> Guardian.validate_admin_entity
+  end
+end
```
Comment (mabiede) [inline pool/web/handler/admin_dashboard.ml:111]:
```suggestion
module Access : sig
  module Statistics : module type of Helpers.Access
end = struct
  module Guardian = Middleware.Guardian

  module Statistics = struct
    include Helpers.Access
    
    let read = Statistics.Guard.Access.read |> Guardian.validate_admin_entity
  end
end
```

File: resources/index.scss [hunk 4]
```diff
@@ -560,3 +560,13 @@ $n: 12;
...
+
+@include until($tablet-breakpoint) {
+    [class*='grid-col-'] [class*='span-'] {
+        grid-column: span 1;
+    }
+}
+
+[class*='grid-col-'].stretch-only-child  > *:only-child{
+    grid-column: 1 / -1;
+}
```
Comment (mabiede) [inline resources/index.scss:572]: missing empty new line

File: pool/app/session/repo/repo.ml (hunk 2 above)
Comment (timohuber) [reply pool/app/session/repo/repo.ml:882]: Currently, I am displaying canceled sessions with a red background, so assistants would see if a session was canceled. We could also add a filter to hide or show them?

Comment (timohuber) [reply pool/app/session/repo/repo.ml:882]: Added legend, as discussed

**Review by mabiede — APPROVED** [review]

**Pull request merged**